pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -335,6 +349,113 @@ func (a *index) serviceEntryWorkloadBuilder(
 	}
 }
 
+func (a *index) endpointSlicesBuilder(
+	MeshConfig krt.Singleton[MeshConfig],
+	WorkloadServices krt.Collection[model.ServiceInfo],
+) krt.TransformationMulti[*discovery.EndpointSlice, model.WorkloadInfo] {
+	return func(ctx krt.HandlerContext, es *discovery.EndpointSlice) []model.WorkloadInfo {
+		// We only care about EndpointSlices that are for a Service.
```

Comment:
Why ? I think ES can be used without Service to represent endpoints that may not have a Service associated ( but something else ?). Don't we want to apply the same rules as we do for pods ?

Reply:
You can represent them, but there is no way for Istio to know what to do with them. They are just data?

Reply:
So from the perspective of 'workload to workload' communication - we treat them as plain text ? 
Instead of for example checking if they have a HBONE port, generating some rule to check spiffe://.../ns/namespace/sa/*, etc ? 

That's related to my security question on how we control malicious IPs - but on the opposite side, assuming ES is strictly locked with RBAC/OPA - and user is using them to represent non-Pod workloads that may or may not use Service but are clearly getting associated with a namespace and have port info ( which in turn may allow inferring if they use HBONE).

Reply:
Also: not sure I fully understand all ambient code, but will this only add endpoints to Service - or will it have any impact on the workload-to-workload communication ?

Reply:
What does it even mean to have an EndpointSlice read by a proxy that does not have a service associated? There is no other proxies doing this (kube-proxy, istio sidecars, etc).

it doesn't have defined semantics.

Reply:
Well - kube-proxy implement Service so I assume watches for ES associated with Service. 

I don't know exactly if istio-sidecars don't 'accidentally' do some label selections or have other means to take EndpointSlices not associated with a Service but with a ServiceEntry or WorkloadGroup or something similar - after all we did in the past use selector-less Services and Endpoints to represent VMs. 

And for 'others' - and semantics - I don't know. It looks like K8S went pretty far in making ES not require a Service - they could have easily added a field and make it required/validated. The semantics are 'a bag of IPs with labels and port info' - just like Gateway API can attach to many things except Service, no clue of ES is not associated with something else, like a ServiceImport or some VM-like resource. 

Not saying I know a specific use - just that we should at least comment the code and make it clear that ES for other things may exist and we choose to ignore them - and perhaps warn if we find them. 

Similar comment for the other things we silently ignore - IPs that are not valid ( hostnames - I have not seen the deprecation announce and what will happen with them), etc. 

Compared with WorkloadEntry - SE provides about the same capabilities but with a more standard interface and some interoperability. Istio is pushing the boundaries with ServiceEntry and WE - we can't expect nobody else does.

That doesn't mean I'm against this change - just trying to understand the risks.

Reply:
> selector-less **Services** and Endpoints to represent VMs.

Key word there is Service... these are ones associated with a service.

In theory we could do something with the arbitrary ones, but I have no clue what. They do not have useful information to us -- no service account, etc.